walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,228 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+    // optionalEvents holds the all of the optionalEvents within the initially given collection of
+    // events
+    private static ArrayList<Event> optionalEvents = new ArrayList<>();
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        // times holds all of the valid time ranges for the given meeting request
+        Collection<TimeRange> times = new ArrayList<>();
+
+        if ((int) request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return times;
+        }
+
+        // If there are no events taking place during the day, the entire day is a valid time range
+        if (events.size() == 0) {
+            times.add(TimeRange.WHOLE_DAY);
+
+            return times;
+        }
+
+        if (events.size() == 1) {
+            boolean included = false;
+
+            for (Event event : events) {
+                if (attends(event, request)) {
+                    included = true;
+                }
+            }
+
+            if (included) {
+                Event[] eventArray = events.toArray(new Event[0]);
+                TimeRange time = eventArray[0].getWhen();
+
+                // Evaluate the time of day before the event's start time
+                if (time.start() - TimeRange.START_OF_DAY >= (int) request.getDuration()) {
+                    times.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, time.start(), false));
+                }
+
+                // Evaluate the time of day after the event's end time
```

Comment:
Please add periods at the end of all of your comments.

Reply:
Ok, I will.